### PR TITLE
Solved the first problem

### PR DESCRIPTION
**Pull request by jabrayare**

### CODE REVIEW

File: Jibril/BinaryTree/week_May_16_22/Leaf_similar_trees.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from typing import Optional
+# Definition for a binary tree node.
+class TreeNode:
+    def __init__(self, val=0, left=None, right=None):
+        self.val = val
+        self.left = left
+        self.right = right
+class Solution:
+    def leafSimilar(self, root1: Optional[TreeNode], root2: Optional[TreeNode]) -> bool:
```
Comment (mohamedsugal) [inline Jibril/BinaryTree/week_May_16_22/Leaf_similar_trees.py:27]: You didn't add iterative solution. If you are having trouble with that I can explain how to do iterative solution on Sunday.

File: Jibril/BinaryTree/week_May_16_22/city_paths.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+
+class TreeNode:
+  def __init__(self, val, distance):
+    self.val = val
+    self.distance = distance
+    self.right = None
+    self.left = None
+
+def path_cities(root, city):
```
Comment (mohamedsugal) [inline Jibril/BinaryTree/week_May_16_22/city_paths.py:24]: You didn't add iterative solution here as well.

File: Jibril/BinaryTree/week_May_16_22/Leaf_similar_trees.py (hunk 1 above)
Comment (jabrayare) [reply Jibril/BinaryTree/week_May_16_22/Leaf_similar_trees.py:27]: Added the iterative solution on this one.

File: Jibril/BinaryTree/week_May_16_22/city_paths.py (hunk 2 above)
Comment (jabrayare) [reply Jibril/BinaryTree/week_May_16_22/city_paths.py:24]: Added the iterative solution on this one but it still has some bugs. I will see if you can find the iterative solution for this one by tomorrow before our session but I would like to see how you went about solving it iteratively.

**Pull request merged**